core/arch/arm/plat-imx/config/config_imx6sx.h

```diff
@@ -36,17 +26,13 @@
  * |        | TA_RAM  |
  * +--------+---------+
  */
-#define CFG_TEE_RAM_PH_SIZE     CFG_TEE_RAM_VA_SIZE
-#define CFG_TEE_RAM_START	TZDRAM_BASE
-#define CFG_TA_RAM_START	ROUNDUP((TZDRAM_BASE + CFG_TEE_RAM_VA_SIZE), \
+#define CFG_TA_RAM_START	ROUNDUP((CFG_TZDRAM_BASE + \
```

Comment:
while you're at it, and if you allow us to take from your time, I saw `CFG_TA_RAM_START`, `CFG_TEE_RAM_START` and `CFG_TEE_RAM_VA_SIZE` do not need to `CFG_` prefix and could move to **platform_config.h** and shared with the few .c, .S source files. This would be a nice cleanup as you want such. 
Once it fine and merge, we will have means to add some clarification entry in the platform port doc. I can take care of this.

Reply:
Let's see if I understand:
 - `CFG_TEE_RAM_PH_SIZE` should be externally configurable. This is the amount of physical memory available to the TEE core.
 - `CFG_TEE_RAM_VA_SIZE` should not be externally configurable, and should be set to `CFG_TEE_RAM_PH_SIZE` rounded up to the next section boundary.
 - `CFG_TA_RAM_START` should not be configurable, and should start at the next section boundary after `CFG_TZDRAM_BASE + CFG_TEE_RAM_VA_SIZE`
 - `CFG_TEE_RAM_START` should not be configurable, and should be set to
     - `CFG_TZDRAM_BASE` if pager not enabled
     - `TZSRAM_BASE` if pager is enabled

Reply:
I've removed `CFG` from `CFG_TA_RAM_START` and `_SIZE`. 

The difficulty with removing `CFG` from `CFG_TEE_RAM_VA_SIZE` is that `CFG_TEE_RAM_PA_SIZE` is usually defined as
```
#define CFG_TEE_RAM_PA_SIZE CFG_TEE_RAM_VA_SIZE
```
So if I make `VA_SIZE` non-configurable, then I also have to make `PA_SIZE` nonconfigurable. I'm still confused what exactly these defines mean. It doesn't make sense to me why `PA_SIZE` is defined as the `VA_SIZE`. I think it would make more sense to supply a value for `PA_SIZE`, and define `VA_SIZE` as the next section size up from `PA_SIZE`. I must be missing something.

The difficulty with removing `CFG` from `CFG_TEE_RAM_START` is that `CFG_TEE_LOAD_ADDR` is usually defined as:
```
#define CFG_TEE_LOAD_ADDR    CFG_TEE_RAM_START
```

So if a remove CFG from START, then I also have to remove it from LOAD_ADDR.